File: etl-tools/src/main/scala/com/ryanquey/intertextualitygraph/models/books/BookRecord.scala
package com.ryanquey.intertextualitygraph.models.books

import scala.beans.BeanProperty;
import com.datastax.oss.driver.api.mapper.annotations.Entity;
import com.datastax.oss.driver.api.mapper.annotations.CqlName;
import com.datastax.oss.driver.api.mapper.annotations.PartitionKey;
import com.datastax.oss.driver.api.mapper.annotations.ClusteringColumn;

import com.ryanquey.datautils.helpers.DataClassesHelpers;
import com.ryanquey.datautils.cassandraHelpers.CassandraDb;


@Entity
@CqlName("books")
class BookRecord (book : Option[Book]) extends BookBase {
  
  @PartitionKey(0) @BeanProperty var name : String // C* TEXT 
  if (book.isDefined) DataClassesHelpers.copyMatchingFields(book, this);

  def getDao () : BookDao = {
    CassandraDb.inventoryMapper.bookDao("books");
  }

  // keeping empty constructor for use with Dao
  def this () = {
    this(null)
  }
};

File: models/src/main/java/com/ryanquey/intertextualitygraph/models/books/BookRecord.java
package com.ryanquey.intertextualitygraph.models.books;

import com.datastax.oss.driver.api.mapper.annotations.Entity;
import com.datastax.oss.driver.api.mapper.annotations.CqlName;
import com.datastax.oss.driver.api.mapper.annotations.PartitionKey;
import com.datastax.oss.driver.api.mapper.annotations.ClusteringColumn;

import com.ryanquey.datautils.helpers.DataClassesHelpers;
import com.ryanquey.datautils.cassandraHelpers.CassandraDb;
import com.ryanquey.datautils.models.Record;
import com.ryanquey.intertextualitygraph.models.InventoryMapper;
import com.ryanquey.intertextualitygraph.models.InventoryMapperObj;

@Entity 
@CqlName("books") 
public class BookRecord extends BookBase implements Record {
  
  @PartitionKey(0) 
  private String name; // C* TEXT 

  BookRecord(Book book) throws Exception {
    DataClassesHelpers.copyMatchingFields(book, this);
  }

  // keeping empty constructor for use with Dao
  BookRecord() {};

  // no sure why there was every nonstatic version
  // TODO DEPRECATED
  BookDao getDao () {
    return InventoryMapperObj
      .inventoryMapper
      .bookDao("books");
  }

  public static BookDao dao () {
    return InventoryMapperObj
      .inventoryMapper
      .bookDao("books");
  }

};


